pebble.go

```diff
@@ -136,6 +136,29 @@ func (db *PebbleDB) DB() *pebble.DB {
 	return db.db
 }
 
+func (db *PebbleDB) Compact(start, end []byte) error {
+	// Currently nil,nil is an invalid range in Pebble.
+	// This was taken from https://github.com/cockroachdb/pebble/issues/1474
+	// In case the start and end keys are the same
+	// pebbleDB will throw an error that it cannot compact.
+	if start != nil && end != nil {
+		return db.db.Compact(start, end, true)
+	}
+
+	iter := db.db.NewIter(nil)
+
+	if start == nil && iter.First() {
+		start = append(start, iter.Key()...)
+	}
+	if end == nil && iter.Last() {
+		end = append(end, iter.Key()...)
+	}
+	if err := iter.Close(); err != nil {
+		return err
+	}
```

Comment:
nit - might be good to leave a comment explaining the rationale for this iteration

Reply:
Line 140 has the link to a comment in pebble's repo where I copied the idea of using the iterator because Pebble does not simply take `Compact(nil,nil)` as an indicator that we want to compact the entire DB.